### PR TITLE
added extra validation for various inputs

### LINKED ISSUE #33: Add date formatting/validation for reservation start dates
When reserving a room, user should not be able to enter an invalid date (i.e december, 3423d, etc.) for the start date

### PR DESCRIPTION
**Pull request by tbuwadi**
added validation for:
- dates in customer new reservation view
- dates in admin new reservation view
- dates in modify reservation view
- cell number in new customer registration view
- credit card number in new customer registration view

changed:
- password field in new customer registration to be *** instead of text

### CODE REVIEW

File: HotelBooking/Views/book_reservation_view.py [hunk 1]
```diff
@@ -4,6 +4,15 @@
 from typing import Tuple, List
 from PyInquirer import prompt
 from HotelBooking.Models.room import ROOM_TYPE
+from datetime import datetime
+
+def ValidateDate(date_text):
```
Comment (mclhtay) [inline HotelBooking/Views/book_reservation_view.py]: can you move this function to Views/utils.py? 
Also, snake case.

Comment (mclhtay) [reply HotelBooking/Views/book_reservation_view.py]: Refactor it to Utils, so the same validation method can be shared by all other views.

Comment (tbuwadi) [reply HotelBooking/Views/book_reservation_view.py]: yessss

File: HotelBooking/Views/reserved_rooms_admin_view.py [hunk 2]
```diff
@@ -5,6 +5,15 @@
 from typing import Tuple, List
 from PyInquirer import prompt
 from HotelBooking.Models.room import ROOM_TYPE
+from datetime import datetime
+
```
Comment (mclhtay) [inline HotelBooking/Views/reserved_rooms_admin_view.py]: Not a comment but a complaint. 

LMAO who named this file? Why does it have such a weird name? Do we have a `reserved_rooms_customer_view` then?

Comment (tbuwadi) [reply HotelBooking/Views/reserved_rooms_admin_view.py]: lolll didn't you make the admin stuff:')

Comment (mclhtay) [reply HotelBooking/Views/reserved_rooms_admin_view.py]: I'm pretty sure I made a file named `reserved_rooms_view`. Someone else split the difference.

Comment (mclhtay) [reply HotelBooking/Views/reserved_rooms_admin_view.py]: Doesn't matter I guess, leave it be.

File: HotelBooking/Views/utils.py [hunk 3]
```diff
@@ -13,3 +13,10 @@ def medium_print(text: str):
 
 def error_print(prefix: str = "", suffix: str = ""):
     print(f'\n{prefix}Please try again{suffix}')
+
+def ValidateDate(date_text):
```
Comment (mclhtay) [inline HotelBooking/Views/utils.py]: Stop it! `validate_date` pls

Comment (tbuwadi) [reply HotelBooking/Views/utils.py]: BE PATIENT ADDING ANOTHER COMMIT RN

File: HotelBooking/Views/utils.py [hunk 4]
```diff
@@ -13,3 +13,10 @@ def medium_print(text: str):
...
+        try:
+            datetime.strptime(date_text, '%m/%d/%Y')
```
Comment (mclhtay) [inline HotelBooking/Views/utils.py:19]: What's the rationale behind testing it as mm/dd/yy? Checking the SQL file it appears we have a different date format?

https://github.com/mclhtay/CS4471/blob/main/HotelBooking/Database/create_database.sql#L89

Comment (mclhtay) [reply HotelBooking/Views/utils.py:19]: Perhaps change the date format in the SQL since you already put mm/dd/yy in a bunch of places.

**Comment by tbuwadi** [comment]
@mclhtay 
added snake case and changed dates in sql and tests to be consistent.. didn't necessarily use same dummy dates but made sure all tests pass

**Review by mclhtay — APPROVED** [review]
Approve on the basis that the author did manual UI testing.

**Pull request merged**